ARCHITECTURE.md

```diff
@@ -1,35 +1,63 @@
 # Architecture
-
-This document describes the high-level architecture of BossRoom.
+This document describes the high-level architecture of Boss Room.
 If you want to familiarize yourself with the code base, you are just in the right place!
 
-> __IMPORTANT__: 
-> This doc is heavily WIP
+Boss Room is an 8-player co-op RPG game experience, where players collaborate to take down some minions, and then a boss. Players can select between classes that each have skills with didactically interesting networking characteristics. Control model is click-to-move, with skills triggered by mouse button or hotkey. 
+
+Code is organized into three separate assemblies: `Client`, `Server` and `Shared` (which, as it's name implies, contains shared functionality that both client and the server require).
+
+## Host model
+Boss Room uses a Host model for its server. This means one client acts as a server and hosts the other clients. 
+
+A common pitfall of this pattern is writing the game in such a way that it is virtually impossible to adapt to a dedicated server model. 
+
+We attempted to combat this by using a compositional model for our client and server logic (rather than having it all combined is single modules):
+ - On the Host, each GameObject has `{Server, Shared, Client}` components. 
+ - If you start up the game as a dedicated server, the client components will disable themselves, leaving you with `{Server, Shared}` components.
+ - If you start up as a client, you get the complementary set of `{Shared, Client}` components. 
+
+This approach works, but requires some care: 
+ - if you have server and clients of a shared base class, you need to remember that the shared code will run twice on the host; 
+ - you also need to take care about code executing in `Start` and `Awake`: if this code runs contemporaneously with the `NetworkingManager`'s initialization, it may not know yet whether the player is a host or client.
```

Comment:
Slight edit
```suggestion
 - Be careful with code executing in `Start` and `Awake`: If this code runs contemporaneously with the `NetworkingManager`'s initialization, it may not know yet if the player is a host or client.
```